settings.gradle.kts

```diff
@@ -106,61 +53,32 @@ project(":clients:osv").name = "osv-client"
-project(":plugins:package-curation-providers:sw360").name = "sw360-package-curation-provider"
-
-project(":plugins:package-managers:bower").name = "bower-package-manager"
-project(":plugins:package-managers:bundler").name = "bundler-package-manager"
-project(":plugins:package-managers:cargo").name = "cargo-package-manager"
-project(":plugins:package-managers:carthage").name = "carthage-package-manager"
-project(":plugins:package-managers:cocoapods").name = "cocoapods-package-manager"
-project(":plugins:package-managers:composer").name = "composer-package-manager"
-project(":plugins:package-managers:conan").name = "conan-package-manager"
-project(":plugins:package-managers:gradle").name = "gradle-package-manager"
-project(":plugins:package-managers:node").name = "node-package-manager"
-project(":plugins:package-managers:nuget").name = "nuget-package-manager"
-project(":plugins:package-managers:pub").name = "pub-package-manager"
-project(":plugins:package-managers:python").name = "python-package-manager"
-project(":plugins:package-managers:spdx").name = "spdx-package-manager"
-project(":plugins:package-managers:stack").name = "stack-package-manager"
-project(":plugins:package-managers:unmanaged").name = "unmanaged-package-manager"
-
-project(":plugins:reporters:asciidoc").name = "asciidoc-reporter"
-project(":plugins:reporters:ctrlx").name = "ctrlx-reporter"
-project(":plugins:reporters:cyclonedx").name = "cyclonedx-reporter"
-project(":plugins:reporters:evaluated-model").name = "evaluated-model-reporter"
-project(":plugins:reporters:fossid").name = "fossid-reporter"
-project(":plugins:reporters:freemarker").name = "freemarker-reporter"
-project(":plugins:reporters:gitlab").name = "gitlab-reporter"
-project(":plugins:reporters:opossum").name = "opossum-reporter"
-project(":plugins:reporters:spdx").name = "spdx-reporter"
-project(":plugins:reporters:static-html").name = "static-html-reporter"
-project(":plugins:reporters:web-app").name = "web-app-reporter"
-
 project(":utils:common").name = "common-utils"
 project(":utils:ort").name = "ort-utils"
 project(":utils:scripting").name = "scripting-utils"
 project(":utils:spdx").name = "spdx-utils"
 project(":utils:test").name = "test-utils"
 
+file("plugins").walk().maxDepth(3).filter {
+    it.isFile && it.name == "build.gradle.kts"
+}.mapTo(mutableListOf()) {
+    it.parentFile.toRelativeString(rootDir).replace(File.separatorChar, ':')
+}.forEach {
+    include(":$it")
+
+    // Give API and package-manager projects a dedicated name that includes the type of plugin.
+    val parts = it.split(':')
+    if (parts.size == 3 && !parts[2].startsWith("gradle-") && !parts[2].endsWith("-template")) {
+        val singularName = parts[1].removeSuffix("s")
+
+        project(":$it").name = if (parts[2] == "api") {
+            "$singularName-api"
+        } else {
+            "${parts[2]}-$singularName"
+        }
+    }
```

Comment:
To me it's hard to grasp what

1. The condition in line 71 is about. Would it make sense to replace it with a hard coded ignore list for
     readability?
2. The reason why the "s" is stripped from the name in line 72

In general, would it make sense to add some code comment before line 62 (stating the purpose of the below code block) and to the logic in the function, to explain a bit the what and why / idea?